nltk/translate/bleu_score.py

```diff
@@ -166,13 +160,12 @@ def corpus_bleu(list_of_references, hypotheses, weights=[0.25, 0.25, 0.25, 0.25]
     # Calculate corpus-level brevity penalty.
     bp = _brevity_penalty(ref_lengths, hyp_lengths)
     
-    # Calculate corpus-level modified precision.
-    p_n = []
-    for i, w in enumerate(weights, start=1):
-        pn = p_numerators[i] / p_denominators[i]
-        p_n.append(w* math.log(pn))
+    # Calculate sum of corpus-level modified precisions.
+    s = (0 if (p_numerators[i] | p_denominators[i]) == 0 else
```

Comment:
I'm not sure whether `0 if not variable else ...` is a good thing, that will allow `None` or "nully` types to pass through. 

```
>>> x, y = None, []
>>> 0 if not(x and y) else x / y
0
>>> x, y = None, 0
>>> 0 if not(x and y) else x / y
0
>>> x, y = 0, 1
>>> 0 if not(x and y) else x / y
0
>>> x, y = 2, ""
>>> 0 if not(x and y) else x / y
0
>>> x, y = 2, 1
>>> 0 if not(x and y) else x / y
2
```